packages/traceloop-sdk/traceloop/sdk/decorators/base.py

```diff
@@ -72,7 +73,13 @@ def wrap(*args, **kwargs):
                 except TypeError as e:
                     Telemetry().log_exception(e)
 
-                res = fn(*args, **kwargs)
+                res = None
+                try:
+                    res = fn(*args, **kwargs)
+
+                except Exception as e:
+                    span.set_status(Status(StatusCode.ERROR))
```

Comment:
You should re-raise the exception otherwise you're hiding it from the user (or not allowing them to handle it themselves)